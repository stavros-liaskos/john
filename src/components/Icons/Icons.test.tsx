import { render } from '@testing-library/react';
import Close from './close';
import Github from './github';
import HandGlass from './handGlass';
import LastFm from './lastfm';
import Moon from './moon';
import Spin from './spin';
import Spotify from './spotify';
import Sun from './sun';

describe('Icons', () => {
  it('matches snapshot', () => {
    const { container, getAllByRole } = render(
      <>
        <Close />
        <Github />
        <HandGlass />
        <LastFm />
        <Moon />
        <Spin />
        <Spotify />
        <Sun />
      </>,
    );

    expect(getAllByRole('img')).toHaveLength(8);
    expect(container).toMatchSnapshot();
  });
});
