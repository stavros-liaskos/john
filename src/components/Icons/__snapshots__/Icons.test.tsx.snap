// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons matches snapshot 1`] = `
<div>
  <svg
    class="h-8 w-8 icon-colour"
    fill="none"
    role="img"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
    viewBox="0 0 24 24"
  >
    <path
      d="M18 6L6 18"
    />
    <path
      d="M6 6L18 18"
    />
  </svg>
  <svg
    class="h-8 w-8 icon-colour"
    fill="none"
    height="24"
    role="img"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M0 0h24v24H0z"
      stroke="none"
    />
    <path
      d="M9 19c-4.286 1.35-4.286-2.55-6-3m12 5v-3.5c0-1 .099-1.405-.5-2 2.791-.3 5.5-1.366 5.5-6.04a4.567 4.567 0 00-1.333-3.21 4.192 4.192 0 00-.08-3.227s-1.05-.3-3.476 1.267a12.334 12.334 0 00-6.222 0C6.462 2.723 5.413 3.023 5.413 3.023a4.192 4.192 0 00-.08 3.227A4.566 4.566 0 004 9.486c0 4.64 2.709 5.68 5.5 6.014-.591.589-.56 1.183-.5 2V21"
    />
  </svg>
  <svg
    class="h-8 w-8 icon-colour"
    fill="none"
    role="img"
    stroke="currentColor"
    viewBox="0 0 24 24"
  >
    <path
      d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
  <svg
    height="40"
    role="img"
    viewBox="0 0 112.195 112.195"
    width="40"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M112.195 56.097c0 30.983-25.114 56.099-56.097 56.099C25.115 112.195 0 87.08 0 56.097 0 25.117 25.115 0 56.099 0c30.982 0 56.096 25.117 56.096 56.097z"
      fill="#d31f27"
    />
    <path
      d="M80.487 52.141c-6.203-1.849-8.412-2.874-8.412-4.922 0-3.416 4.937-4.702 5.497-4.79 3.139-.465 7.072 1.173 8.34 4.99l8.44-2.553c-2.65-8.385-10.487-12.099-18.099-10.975-7.788 1.146-12.782 6.786-12.782 13.327 0 8.8 8.244 11.332 14.43 13.181 6.509 1.944 8.401 2.637 8.401 5.023 0 2.016-1.196 3.895-4.374 4.735-6.32 1.664-14.586-.672-16.887-4.523-1.351-2.261-2.733-5.203-4.001-8.53-3.766-9.866-8.926-23.371-23.764-23.371-7.731 0-21.851 3.743-21.851 24.868 0 8.744 7.481 19.613 21.578 19.613 11.829 0 13.888-4.32 14.358-5.085L47.5 65.574c-.098.138-3.692 4.936-10.497 4.936-11.107 0-12.735-11.776-12.735-11.909 0-10.967 4.687-16.238 13.008-16.238 8.094 0 11.382 7.04 15.48 17.759 1.397 3.683 2.922 6.966 4.65 9.865 3.657 6.116 11.55 8.091 19.473 8.091 2.418 0 5.2.137 7.351-.432 7.093-1.878 10.905-6.178 10.905-12.224.002-9.204-8.283-11.378-14.648-13.281z"
      fill="#fff"
    />
  </svg>
  <svg
    class="h-8 w-8 text-gh-dark"
    fill="none"
    role="img"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
    viewBox="0 0 24 24"
  >
    <path
      d="M21 12.79A9 9 0 1111.21 3 7 7 0 0021 12.79z"
    />
  </svg>
  <svg
    class="animate-spin text:black dark:text-white"
    fill="none"
    height="15"
    role="img"
    viewBox="0 0 24 24"
    width="15"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      class="opacity-25"
      cx="12"
      cy="12"
      r="10"
      stroke="currentColor"
      stroke-width="4"
    />
    <path
      class="opacity-75"
      d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
      fill="currentColor"
    />
  </svg>
  <svg
    height="40"
    role="img"
    viewBox="0 0 128 128"
    width="40"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      clip-rule="evenodd"
    >
      <path
        d="M0 0h128v128H0z"
        fill="none"
      />
      <path
        d="M64 0C28.8 0 0 28.8 0 64s28.8 64 64 64 64-28.8 64-64S99.52 0 64 0m29.44 92.48c-1.28 1.92-3.52 2.56-5.44 1.28-15.04-9.28-33.92-11.2-56.32-6.08-2.24.64-4.16-.96-4.8-2.88-.64-2.24.96-4.16 2.88-4.8 24.32-5.44 45.44-3.2 62.08 7.04 2.24.96 2.56 3.52 1.6 5.44m7.68-17.6c-1.6 2.24-4.48 3.2-6.72 1.6-17.28-10.56-43.52-13.76-63.68-7.36-2.56.64-5.44-.64-6.08-3.2-.64-2.56.64-5.44 3.2-6.08 23.36-7.04 52.16-3.52 72 8.64 1.92.96 2.88 4.16 1.28 6.4m.64-17.92C81.28 44.8 47.04 43.52 27.52 49.6c-3.2.96-6.4-.96-7.36-3.84-.96-3.2.96-6.4 3.84-7.36 22.72-6.72 60.16-5.44 83.84 8.64 2.88 1.6 3.84 5.44 2.24 8.32-1.6 2.24-5.44 3.2-8.32 1.6"
        fill="#10BC4C"
        fill-rule="evenodd"
      />
    </g>
  </svg>
  <svg
    class="h-8 w-8 text-slate-300"
    fill="none"
    role="img"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
    viewBox="0 0 24 24"
  >
    <circle
      cx="12"
      cy="12"
      r="5"
    />
    <path
      d="M12 1L12 3"
    />
    <path
      d="M12 21L12 23"
    />
    <path
      d="M4.22 4.22L5.64 5.64"
    />
    <path
      d="M18.36 18.36L19.78 19.78"
    />
    <path
      d="M1 12L3 12"
    />
    <path
      d="M21 12L23 12"
    />
    <path
      d="M4.22 19.78L5.64 18.36"
    />
    <path
      d="M18.36 5.64L19.78 4.22"
    />
  </svg>
</div>
`;
